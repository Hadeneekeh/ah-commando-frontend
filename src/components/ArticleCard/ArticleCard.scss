$desktop: 376px;

%svg {
  width: 18.37px;
  height: 14px;
  color: rgba(0, 0, 0, 0.6);
  cursor: pointer;
}
%svg-horizontal {
  width: 11.12px;
  height: 11px;
  margin-left: 5px;
  cursor: pointer;
}
%labels {
  font-family: 'Roboto', sans-serif;
  display: flex;
}

@mixin desktop {
   @media (min-width: #{$desktop}) {
     @content
   }
}

.vertical {
  width: 115.72px;
  height: 193px;
  margin: 10px 20px 10px 0;
  @include desktop {
    width: 193px;
    height: 250px;
  }
  .image-container {
    width: 100%;
    margin: 0 auto;
    @include desktop{
      height: 129px;
    }
  }
  .img {
    width: 115.72px;
    height: 79.98px;
    margin: 0 auto;
    object-fit: cover;
    border-radius: 5px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    @include desktop{
      width: 193px;
      height: 129px;
    }
  }

  .article-details {
    .title-container {
      .title {
        @extend %labels;
        font-family: 'Montserrat', sans-serif;
        font-weight: 500;
        font-size: 11px;
        margin-top: 10px;
        @include desktop{
          font-size: 12px;
          overflow: hidden;
          text-overflow: ellipsis;
          display: block;
          white-space: nowrap;
          width: 90%;
        }
      }
      .author-name {
        @extend %labels;
        text-transform: capitalize;
        font-weight: 300;
        font-size: 11px;
        margin-top: 5px;
        @include desktop{
          font-size: 12px;
        }
      }
      .description {
        display: none;
      }
    }
    .icons {
      display: flex;
      flex-direction: row;
      align-items: center;
      margin-top: 15px;
      color:rgba(0, 0, 0, 0.8);
      cursor: pointer;
      .like, .dislike, .comment{
        display: flex;
        align-items: center;
        font-size: 10px;
        @include desktop {
          font-size: 14px;
        }
      }

      .comment, .dislike {
        margin-left: 5px;
        font-size: 10px;
      }

      .icon-label {
        @extend %labels;
        font-family: 'Monaco', sans-serif;
        font-size: 12px;
        height: 100%;
        color: rgba(0, 0, 0, 0.7);
        position: relative;
        right: 9px;
        @include desktop {
          font-size: 12px;
        }
      }
      .icon-container {
        font-size: 11px;
        color: rgba(0, 0, 0, 0.7);
      }
      .hor-readTime {
        display: none;
      }
    }
    .ver-readTime {
      @extend %labels;
      font-family: 'Montserrat', sans-serif;
      font-weight: 900;
      font-size: 10px;
      margin-top: 10px;
    }
  }
}

.horizontal {
  width: 355.5px;
  height: 102.86px;
  display: grid;
  grid-template-columns: 27.94% 72.06%;
  grid-template-areas: 'image body';
  @include desktop {
    width: 986px;
    height: 147px;
  }
  .image-container {
    width: 100%;
    margin: 0 auto;
    grid-area: image;
    @include desktop{
      width: 100%;
      margin: 0 auto;
      height: 147px;
    }
  }
  .img {
    width: 88.29px;
    height: 64.31px;
    margin: 0 auto;
    border-radius: 5px;
    object-fit: cover;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    transition: all 1s ease-in;
    @include desktop{
      width: 100%;
      height: 147px;
    }
  }
  .article-details {
    grid-area: body;
    padding: 2px;
    @include desktop{
      padding: 0 174px 0 20px;
    }
    .title-container {
      .title {
        @extend %labels;
        font-family: 'Montserrat', sans-serif;
        font-size: 10px;
        margin: 3px 3px;
        @include desktop{
          font-size: 15px;
          margin: 15px 0 3px 0;
        }
      }
      .author-name {
        @extend %labels;
        text-transform: capitalize;
        font-weight: 300;
        font-size: 8px;
        margin: 3px 3px;
        @include desktop{
          font-size: 14px;
          margin: 5px 3px;
        }
      }
      .description {
        @extend %labels;
        display: -webkit-box;
        -webkit-line-clamp: 2;
        -webkit-box-orient: vertical;
        overflow: hidden;
        text-overflow: ellipsis;
        font-size: 9px;
        margin: 3px 3px;
        font-weight: 100;
        @include desktop{
          font-family: 'Montserrat', sans-serif;
          font-size: 14px;
          margin: 5px 3px;
        }
      }
    }
    .icons {
      display: flex;
      flex-direction: row;
      align-items: center;
      cursor: pointer;
      margin-top: 5px;
      margin: 3px;
      @include desktop {
        margin-top: 15px;
      }
      .like, .dislike,  .comment{
        display: flex;
        align-items: center;
        z-index: 1;
      }

      .dislike {
        margin-left: 8px;
        @include desktop {
          margin-left: 15px;
        }
      }

      .comment {
        margin-left: 8px;
        @include desktop {
          margin-left: 30px;
        }
      }
      .icon-label {
        @extend %labels;
        font-family: 'Monaco';
        font-size: 12px;
        line-height: 13px;
        color: rgba(0, 0, 0, 0.7);
        margin-left: 3px;
        position: relative;
        right: 11px;
      }

      .icon-container {
        font-size: 11px;
        color: rgba(0, 0, 0, 0.7);
      }

      .hor-readTime {
        display: none;
        @include desktop {
          font-family: 'Montserrat', sans-serif;
          font-weight: 900;
          font-size: 10px;
          display: block;
          font-size: 10px;
          margin-left: auto;
        }
      }
    }
    .ver-readTime {
      display: block;
      font-family: 'Montserrat', sans-serif;
      font-weight: 900;
      font-size: 10px;
      margin-top: 5px;
      @include desktop {
        display: none;
      }
    }
  }
}
