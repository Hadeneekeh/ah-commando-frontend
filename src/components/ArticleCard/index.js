/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { PropTypes } from 'prop-types';
import './ArticleCard.scss';
import Icon from '../Icon';

const ArticleCard = (props) => {
  const { type, data } = props;
  return (
    <div className={`${type}`}>
      <div className="image-container">
        <img className="img" alt="logo" src={data.image} />
      </div>
      <div className="article-details">
        <div className="title-container">
          <label className="title">{data.title}</label>
          <label className="author-name">{data.authorName}</label>
          <label className="description">{data.description}</label>
        </div>
        <div className="icons">
          <div className="like">
            <Icon name="likes" className="like" />
            <label className="icon-label">{data.likes}</label>
          </div>
          <div className="dislike">
            <Icon name="dislikes" />
            <label className="icon-label">{data.dislikes}</label>
          </div>
          <div className="comment">
            <Icon name="comments" />
            <label className="icon-label">{data.comments}</label>
          </div>
          <label className="hor-readTime">{`${data.readTime} min read`}</label>
        </div>
        <label className="ver-readTime">{`${data.readTime} min read`}</label>
      </div>
    </div>
  );
};

ArticleCard.propTypes = {
  type: PropTypes.string,
  data: PropTypes.shape({
    image: PropTypes.string,
    title: PropTypes.string,
    authorName: PropTypes.string,
    description: PropTypes.string,
    likes: PropTypes.number,
    dislikes: PropTypes.number,
    comments: PropTypes.number,
    readTime: PropTypes.number,
  }),
};

ArticleCard.defaultProps = {
  type: '',
  data: {
    image: '',
    title: '',
    authorName: '',
    description: '',
    likes: 2,
    dislikes: 1,
    comments: 1,
    readTime: 1,
  },
};

export default ArticleCard;
