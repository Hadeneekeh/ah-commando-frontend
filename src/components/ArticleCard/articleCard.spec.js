import { shallow } from 'enzyme';
import React from 'react';
import findByTestAttribute from '@Utils/';
import ArticleCard from './';

const setUp = props => shallow(<ArticleCard {...props} />);

describe('Article card component test', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setUp();
  });
  describe('Card tests', () => {
    it('Should render without failing', () => {
      const component = findByTestAttribute(wrapper, 'articleCardComponent');
      expect(component.length).toBe(1);
    });
  });
});
