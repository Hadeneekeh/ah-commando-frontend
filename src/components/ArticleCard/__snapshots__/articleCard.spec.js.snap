// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component test test like or dislike and article should like an article 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "",
      21,
      "article",
    ],
    Array [
      "",
      21,
      "article",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": false,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
